doc: Expand section on Getting Started (#19072)
bitcoin/bitcoin: CONTRIBUTING.md

merging pull requests as well as a "lead maintainer" who is responsible for the
release cycle, overall merging, moderation and appointment of maintainers.

Getting Started
---------------

> New developers are very welcome and needed. There are a lot of open issues of

Reviewer: suggest s/a lot/many/

Reviewer: It may be a good idea to work in a phrase, similar to this one, early in this section: 

Reviewing and testing is the most effective way you can contribute as a new contributor, and also will teach you much more about the code and process than opening PRs.

> any difficulty waiting to be fixed. However, before you start contributing,

Reviewer 2: ```suggestion
varying difficulty waiting to be fixed. However, before you start contributing,
```

familiarize yourself with the Bitcoin Core build system and tests. Refer to the
documentation in the repository on how to build Bitcoin Core and how to run the
unit and functional tests.

If you're looking for somewhere to start contributing, check out the
[good first issue](https://github.com/bitcoin/bitcoin/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
list or changes that are
[up for grabs](https://github.com/bitcoin/bitcoin/issues?utf8=%E2%9C%93&q=label%3A%22Up+for+grabs%22).
Some of them might no longer be applicable. So if you are interested, but
unsure, you might want to leave a comment on the issue first.

You may also participate in the weekly
[Bitcoin Core PR Review Club](https://bitcoincore.reviews/) meeting.

> Finally, review and testing is important as well. It gives you an idea of what

Reviewer 2: ```suggestion
Finally, review and testing are important as well. It gives you an idea of what
```

others are working on, where you can help, and how the review process works.
Experience from testing and reviewing will help you become a better
contributor. Please refer to the section [peer review](#peer-review) later in
this document.

### Good First Issue Label

The purpose of the `good first issue` label is to highlight which issues are suitable for a new contributor without a deep understanding of the codebase.

> However, `good first issue`s can be solved by anyone. If they remain unsolved for a longer time, a common contributor might address them.

Reviewer 2: consider frequent for common to clarify

> You do not need to request permission to start working on this. You are encouraged to comment on the issue if you are planning to work on it. This will help other contributors monitor which issues are actively being addressed and is also an effective way to request assistance if and when you need it.

Reviewer 2: ```suggestion
You do not need to request permission to start working on an issue. However, you are encouraged to leave a comment if you are planning to work on it. This will help other contributors monitor which issues are actively being addressed and is also an effective way to request assistance if and when you need it.
```

merging pull requests as well as a "lead maintainer" who is responsible for the
release cycle, overall merging, moderation and appointment of maintainers.

Getting Started
---------------

> New developers are very welcome and needed.

Reviewer: s/developers/contributors/?

Reviewing and testing is the most effective way you can contribute as a new

> contributor, and also will teach you much more about the code and process than

Reviewer: s/and/and it/

> opening PRs. Please refer to the section [peer review](#peer-review) later in

Reviewer: `[Peer Review](#peer-review)`

this document.

Before you start contributing, familiarize yourself with the Bitcoin Core build
system and tests. Refer to the documentation in the repository on how to build
Bitcoin Core and how to run the unit and functional tests.

> There are many of open issues of varying difficulty waiting to be fixed.

Reviewer: s/many of/many/

If you're looking for somewhere to start contributing, check out the
[good first issue](https://github.com/bitcoin/bitcoin/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
list or changes that are
[up for grabs](https://github.com/bitcoin/bitcoin/issues?utf8=%E2%9C%93&q=label%3A%22Up+for+grabs%22).
Some of them might no longer be applicable. So if you are interested, but
unsure, you might want to leave a comment on the issue first.

You may also participate in the weekly
[Bitcoin Core PR Review Club](https://bitcoincore.reviews/) meeting.

### Good First Issue Label

The purpose of the `good first issue` label is to highlight which issues are
suitable for a new contributor without a deep understanding of the codebase.

> However, `good first issue`s can be solved by anyone. If they remain unsolved

Reviewer: `good first issue`s isn't very readable; suggest plain text here: "good first issues"

The developer
[mailing list](https://lists.linuxfoundation.org/mailman/listinfo/bitcoin-dev)

> should be used to discuss complicated or controversial consensus or protocol changes before working on

Reviewer 3: maybe `P2P protoocol`? (to distinguish it from internal or interprocess interfaces)

Reviewer 4: Done. Fixed in commit 100000d1

The project coding conventions in the [developer notes](doc/developer-notes.md)
must be followed.

> ### Commiting Patches

Reviewer: Committing

    qt: Add feed bump button
    log: Fix typo in log message

The body of the pull request should contain enough description about what the
patch does together with any justification/reasoning. You should include

> references to any discussions (for example other tickets or mailing list

Reviewer 2: ```suggestion
references to any discussions (for example, other tickets or mailing list
```

    qt: Add feed bump button
    log: Fix typo in log message

The body of the pull request should contain enough description about what the
patch does together with any justification/reasoning. You should include
references to any discussions (for example other tickets or mailing list
discussions).

### Translation changes

Note that translations should not be submitted as pull requests. Please see
[Translation Process](https://github.com/bitcoin/bitcoin/blob/master/doc/translation_process.md)
for more information on helping with translations.

> ### Work in Progress changes and Requests for Comment

Reviewer: s/Comment/Comments/ or Feedback (in singular form)
